### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by AyusheGangal**
Frontend PR

### CODE REVIEW

File: view/gui.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    def generate_and_show_gui(self):
+        with st.sidebar:
+            st.markdown("<h1 style='text-align: center;font-size:100px;'>🧭</h1>", unsafe_allow_html=True)
+            st.markdown("<h1 style='text-align: center;'>EleNa: Elevation Based Navigation System</h1>", unsafe_allow_html=True)
+
+            BASEMAPS = self.config['BASEMAPS']
+            TRAVEL_MODE = self.config['TRAVEL_MODE']
+            ADDRESS_DEFAULT = self.config['ADDRESS_DEFAULT']
+
+            basemap = st.selectbox("Select Map Type", BASEMAPS)
```
Comment (muditchaudhary) [inline view/gui.py]: Can be changed to `basemap = st.selectbox("Select Map Type", self.config["BASEMAPS"])`. Same for other attributes mentioned in the config.

File: view/launch.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import gui
+
+if __name__ == "__main__":
+    GUI = gui() 
+    GUI.to_streamlit()
```
Comment (muditchaudhary) [inline view/launch.py]: I think it should be` GUI.generate_and_show_gui()`. 
`.to_streamlit()` was to add the map iirc.

File: view/gui.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            st.write("Shows path upto "+ str(tolerance) +" times the shortest path")
+            
+            col1, col2 = st.columns([1,1])
+            with col1:
+                st.button("Clear Choice", on_click=clear_text)
+            with col2:
+                st.button("Get Directions", on_click=get_directions)
+        
+    def get_directions():
+        pass
```
Comment (muditchaudhary) [inline view/gui.py]: Add input validation --> popup/display with message on incorrect input.

For now if input validation passes --> popup/display confirmation that input validation is passed. Will modify it to actual implementation later.

**Review by muditchaudhary — CHANGES_REQUESTED** [review]

File: view/gui.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+class gui:
+    def __init__(self, filename = "config.json"):
+        with open(filename) as config_file:
+            self.config = json.load(config_file)
+    
+    def clear_text(self):
+        st.session_state["go_from"] = ""
+        st.session_state["go_to"] = ""
+
+    def generate_and_show_gui(self):
```
Comment (muditchaudhary) [inline view/gui.py]: It seems to be missing the initial map being displayed. For now, set the longitude latitude of the map to be 


```
lat, lon = 42.4529, -72.5653

m = leafmap.Map(center=(lat, lon), zoom=16)

m.add_basemap(basemap)

m.to_streamlit()
```

**Review by muditchaudhary — CHANGES_REQUESTED** [review]

File: view/gui.py (hunk 1 above)
Comment (AyusheGangal) [reply view/gui.py]: Done.

File: view/launch.py (hunk 2 above)
Comment (AyusheGangal) [reply view/launch.py]: Done

File: view/gui.py (hunk 4 above)
Comment (AyusheGangal) [reply view/gui.py]: Done

File: view/gui.py (hunk 3 above)
Comment (AyusheGangal) [reply view/gui.py]: Done

File: view/navigator_demo.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import networkx as nx
```
Comment (muditchaudhary) [inline view/navigator_demo.py:1]: This is for demo only

**Review by muditchaudhary — APPROVED** [review]
Refined GUI | Added a demo integration for UI testing

**Pull request merged**